### PR TITLE
Impove integration-test suite

### PR DESCRIPTION
**Pull request by ahmedsmiai**

### CODE REVIEW

File: src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt [hunk 1]
```diff
@@ -17,5 +20,93 @@ class Erc721StandardTest : StringSpec({
...
+
+    "update meta data".config(enabled=enableTest){
+        val result = NftService.updateMetadata(chain=Chain.MUMBAI,
+            contractAddress = "0xf277BE034881eE38A9b270E5b6C5c6f333Af2517",
+            tokenId="4",
+            key= "string",
+            value= "string",
+            signedAccount = null
+            )
+        result.transactionId shouldNotBe null
```
Comment (philpotisk) [inline src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt:87]: To check if an update has succeeded, there should be always a read-request to validate the data.

File: src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt [hunk 2]
```diff
@@ -17,5 +20,93 @@ class Erc721StandardTest : StringSpec({
...
+        result.name shouldBe "string"
+        result.image shouldBe "string"
+        result.external_url shouldBe "string"
+        result.attributes?.get(0)!!.trait_type shouldBe "string"
+        result.attributes?.get(0)!!.value shouldBe "string"
+
+    }
+
+    "add file to ipfs nft storage".config(enabled=enableTest){
+        val byteArray = Files.readAllBytes(Paths.get("C:\\Users\\ahmed\\Desktop\\sa7eHXGY_400x400.jpg"))
```
Comment (philpotisk) [inline src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt:105]: this should not be hardcoded

**Review by philpotisk — CHANGES_REQUESTED** [review]
There are several failing tests. please make sure that they run reliable.
image
Please go over the code and discuss each test-case with @khemiriwalid

File: src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt [hunk 3]
```diff
@@ -17,5 +24,105 @@ class Erc721StandardTest : StringSpec({
...
+        result.description shouldBe "Ticket #1 Description"
+
+    }
+
+    "get metadata uri".config(enabled=enableTest){
+        val tokenid = BigInteger.valueOf(4)
+        val result = NftService.getNftMetadataUri(Chain.MUMBAI,
+            "0xf277BE034881eE38A9b270E5b6C5c6f333Af2517", tokenid )
+        result shouldNotBe null
+        result shouldNotBe ""
```
Comment (khemiriwalid) [inline src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt:55]: You need to compare the return value from the "getNftMetadataUri" function  with the expected result.

File: src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt [hunk 4]
```diff
@@ -17,5 +24,105 @@ class Erc721StandardTest : StringSpec({
...
+    }
+
+    "get owner of a token test".config(enabled=enableTest){
+        val tokenid = BigInteger.valueOf(4)
+
+        val result = NftService.ownerOf(Chain.MUMBAI,
+            "0xf277BE034881eE38A9b270E5b6C5c6f333Af2517",
+        tokenid)
+        result shouldNotBe null
+        result shouldNotBe ""
```
Comment (khemiriwalid) [inline src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt:72]: You need to compare the return value from the "ownerOf" function with the expected result.

File: src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt [hunk 5]
```diff
@@ -17,5 +24,105 @@ class Erc721StandardTest : StringSpec({
...
+    }
+
+    "get token collection info".config(enabled=enableTest){
+        val result = NftService.getTokenCollectionInfo(Chain.MUMBAI,
+            "0xf277BE034881eE38A9b270E5b6C5c6f333Af2517" )
+        result.name shouldNotBe null
+        result.name shouldNotBe ""
```
Comment (khemiriwalid) [inline src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt:79]: You need to compare the return value from the "getTokenCollectionInfo" function with the expected result.

File: src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt [hunk 6]
```diff
@@ -17,5 +24,105 @@ class Erc721StandardTest : StringSpec({
...
+    }
+
+    "get account nfts by alchmy".config(enabled=enableTest){
+        val result = NftService.getAccountNFTsByAlchemy(Chain.MUMBAI,
+            "0xf277BE034881eE38A9b270E5b6C5c6f333Af2517" )
```
Comment (khemiriwalid) [inline src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt:84]: Put an account address in the second parameter of the "getAccountNFTsByAlchemy" function

File: src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt [hunk 7]
```diff
@@ -2,19 +2,133 @@ package id.walt.nftkit.erc721Features
...
+    "Smart contract deployment".config(enabled = true) {
         val deploymentOptions = DeploymentOptions(AccessControl.OWNABLE, TokenStandard.ERC721)
         val deploymentParameter = DeploymentParameter("Metaverse", "MTV", DeploymentParameter.Options(true, true))
         val result = NftService.deploySmartContractToken(Chain.MUMBAI, deploymentParameter, deploymentOptions)
+        val privateKey= WaltIdServices.loadChainConfig().privateKey
+        val credentials: Credentials = Credentials.create(privateKey)
+
         result.contractAddress shouldNotBe  ""
         result.contractAddress shouldNotBe null
+
```
Comment (khemiriwalid) [inline src/test/kotlin/id/walt/nftkit/erc721Features/Erc721StandardTest.kt:29]: We can call "getTokenCollectionInfo" function to check the name and symbol of the deployed smart contract

File: src/test/kotlin/id/walt/nftkit/accessControl/EvmBasedAccessControl.kt [hunk 8]
```diff
@@ -10,11 +10,53 @@ class EvmBasedAccessControl : StringSpec({
...
+    "transfer ownership".config(enabled=enableTest){
+        val result = AccessControlService.transferOwnership(Chain.MUMBAI,
+            "0x41ffba64969ab4d8a4ac7fae765e4fc1dd30290b",
+            "0xaf87c5ce7a1fb6bd5aadb6dd9c0b8ef51ef1bc31"
+        )
+        val nOwner= AccessControlService.owner(Chain.MUMBAI, "0x41ffba64969ab4d8a4ac7fae765e4fc1dd30290b")
+        nOwner shouldBe "0xaf87c5ce7a1fb6bd5aadb6dd9c0b8ef51ef1bc31"
+        result.transactionId shouldNotBe null
+    }
+
```
Comment (khemiriwalid) [inline src/test/kotlin/id/walt/nftkit/accessControl/EvmBasedAccessControl.kt:27]: Add new test case for "Renounce ownership".
1- Deploy new smart contract
2- Call the "renounceOwnership" function on that smart contract instance
3- Check the owner of the smart contract.

File: src/test/kotlin/id/walt/nftkit/accessControl/EvmBasedAccessControl.kt [hunk 9]
```diff
@@ -10,11 +10,53 @@ class EvmBasedAccessControl : StringSpec({
...
+    " grant role ".config(enabled=enableTest){
+        val result = AccessControlService.grantRole(Chain.MUMBAI,"0xa5a0914988bAB4e773109969A9176855eA77FcfB", role="MINTER_ROLE" , "0xaf87c5ce7a1fb6bd5aadb6dd9c0b8ef51ef1bc31" )
+        val has = AccessControlService.hasRole(Chain.MUMBAI, "0xa5a0914988bAB4e773109969A9176855eA77FcfB", "MINTER_ROLE" , "0xaf87c5ce7a1fb6bd5aadb6dd9c0b8ef51ef1bc31")
+        result.transactionId shouldNotBe null
+        has shouldBe true
```
Comment (khemiriwalid) [inline src/test/kotlin/id/walt/nftkit/accessControl/EvmBasedAccessControl.kt:32]: Call the "hasRole" function to checkout if the account has the specified role

**Review by khemiriwalid — APPROVED** [review]

**Comment by philpotisk** [comment]
nice one :+1:

**Pull request merged**